core/src/autogluon/core/_setup_utils.py

```diff
@@ -19,7 +19,7 @@
     "boto3": ">=1.10,<2",  # <2 because unlikely to introduce breaking changes in minor releases. >=1.10 because 1.10 is 3 years old, no need to support older
     "numpy": ">=1.21,<1.29",  # "<{N+3}" upper cap, where N is the latest released minor version, assuming no warnings using N
     "pandas": ">=2.0.0,<2.3.0",  # "<{N+3}" upper cap
-    "scikit-learn": ">=1.3.0,<1.4.1",  # "<{N+1}" upper cap, capping to micro version as AutoMM HPO tests fail on upgrading to higher version
+    "scikit-learn": ">=1.4.0,<1.5.2",  # capping to latest version
```

Comment:
This is worthy of running the tabular benchmark, to ensure that there isn't a regression.

Reply:
Will run a benchmark today!

Reply:
[scikit_results_automlbenchmark_1h_ag_bench_debug_20240828T184249.csv](https://github.com/user-attachments/files/16792588/scikit_results_automlbenchmark_1h_ag_bench_debug_20240828T184249.csv)

Attaching raw results for this